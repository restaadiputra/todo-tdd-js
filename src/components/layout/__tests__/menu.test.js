import React from 'react';
import Menu from '../menu';
import { renderWithRouter } from 'test/utils';

test('contain hidden class when menu is not open', () => {
  const { container } = renderWithRouter(<Menu openMenu={false} />, {});
  expect(container.firstChild.getAttribute('class')).toContain('max-h-0');
  expect(container.firstChild).toMatchSnapshot();
});

test('not contain hidden class when menu is open', () => {
  const { container } = renderWithRouter(<Menu openMenu={true} />, {});
  expect(container.firstChild.getAttribute('class')).toContain('max-h-200');
  expect(container.firstChild).toMatchSnapshot();
});
