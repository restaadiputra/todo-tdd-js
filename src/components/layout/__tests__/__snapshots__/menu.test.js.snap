// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contain hidden class when menu is not open 1`] = `
<div
  class="w-full lg:inline-flex lg:flex-grow lg:w-auto hidden"
  data-testid="menu"
>
  <div
    class="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
  >
    <a
      class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white"
      href="/props"
    >
      <span>
        Props
      </span>
    </a>
    <a
      class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white"
      href="/context"
    >
      <span>
        Context
      </span>
    </a>
    <a
      class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white"
      href="/redux"
    >
      <span>
        Redux
      </span>
    </a>
  </div>
</div>
`;

exports[`not contain hidden class when menu is open 1`] = `
<div
  class="w-full lg:inline-flex lg:flex-grow lg:w-auto"
  data-testid="menu"
>
  <div
    class="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
  >
    <a
      class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white"
      href="/props"
    >
      <span>
        Props
      </span>
    </a>
    <a
      class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white"
      href="/context"
    >
      <span>
        Context
      </span>
    </a>
    <a
      class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white"
      href="/redux"
    >
      <span>
        Redux
      </span>
    </a>
  </div>
</div>
`;
